client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -59,6 +73,12 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goToStep, flow }: SidebarPr
 	const [ showConfirmModal, setShowConfirmModal ] = useState( false );
 	const queryClient = useQueryClient();
 
+	useEffect( () => {
+		if ( showConfirmModal ) {
+			recordUnverifiedDomainDialogShownTracksEvent( site?.ID );
+		}
+	}, [ site, showConfirmModal ] );
```

Comment:
I haven't tested but noted this:

I find this sort of thing, tracking prop changes using `useEffect`, a bit hacky. The tracks event should only ever be fired on click, but there could be (hypothetically I guess), some other thing down the track that changes the value of the prop too. I think it's better to tie tracks events more directly to the event they're supposed to represent.

Looks like we could instead fire the event in the callback that's passed to `getEnhancedTasks()`. Like

```ts
		getEnhancedTasks(
			launchpadChecklist,
			siteSlug,
			site,
			submit,
			globalStylesInUse && shouldLimitGlobalStyles,
			PLAN_PREMIUM,
			setShowPlansModal,
			queryClient,
			goToStep,
			flow,
			isEmailVerified,
			checklistStatuses,
			getPlanCartItem(),
			getDomainCartItem(),
			stripeConnectUrl,
			() => {
				recordUnverifiedDomainDialogShownTracksEvent( site?.ID );
				setShowConfirmModal( true );
			},
			isDomainEmailUnverified
		);
```

Reply:
Good one. I updated it,.